src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java

```diff
@@ -95,12 +104,13 @@ public ResponseEntity<Object> updateDriverAvailability(
                             @RequestParam Long id,
                             @RequestBody @Valid DriverAvailability incoming)
     {
-        DriverAvailability availability;
+        DriverAvailability availability = driverAvailabilityRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(DriverAvailability.class, id));
 
-        availability = driverAvailabilityRepository.findById(id)
-                    .orElseThrow(() -> new EntityNotFoundException(DriverAvailability.class, id));
+        if (availability.getDriverId() != getCurrentUser().getUser().getId()) {
```

Comment:
Member
@[pconrad](https://github.com/pconrad) pconrad [Jun 4, 2024](https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-5/pull/37/files#r1626804244)

This is not the correct exception to throw. This should only be used when you did a search for a DriverAvailability with a certain id and it was not found. Throwing this exception will cause future devs to curse your name.

The exception you want is one that will turn into a 403 unauthorized, but if you can't figure out how to do that, then just throw a generic exception with a reasonable error message.
@pconrad